### PR TITLE
Sub label api score

### PR DESCRIPTION
**Pull request by NickM-27**
Enabled the sub_label API to include setting the sub label score and also fixes the case where only a string was being used for sub labels

### CODE REVIEW

**Comment by skrashevich** [comment]
What type is `subLabelScore` expected to have?

**Comment by NickM-27** [comment]
> What type is `subLabelScore` expected to have?

It's a float like it shows in the API docs I updated

**Comment by skrashevich** [comment]
After merging #7100, in some cases, in sub_label saved only first character of sublabel. Unfortunately, I have not yet been able to find the pattern and give steps for reproduce.

**Comment by NickM-27** [comment]
> After merging #7100, in some cases, in sub_label saved only first character of sublabel. Unfortunately, I have not yet been able to find the pattern and give steps for reproduce.

this PR fixes that

**Comment by skrashevich** [comment]
Perhaps it makes sense to add an attribute like "source", to be able to switch from frigate events page to the external tool (application), that generated the sublabel?

**Comment by NickM-27** [comment]
I'm not sure if that's something that makes sense. 

There's no way to know where that other service is hosted given that docker ports are customizable so that would have to be configured. There's added complexity in cases where different custom domains are used as well. Finally, it would be impossible as is to make it work in the addon (see issues with go2rtc dashboard in the addon).

**Comment by skrashevich** [comment]
image
in database (and request) score is 91.47

**Comment by NickM-27** [comment]
> image
> in database (and request) score is 91.47

It's supposed to be a float < 1 like the docs I've put here show. This is consistent with the rest of the codebase as well 👍

**Comment by skrashevich** [comment]
maybe, return an error if `float(subLabelScore)` < 0 or > 1 ?

File: frigate/http.py [hunk 1]
```diff
@@ -386,6 +386,18 @@ def set_sub_label(id):
             400,
         )
 
+    if new_score > 1.0 or new_score < 0:
+        return make_response(
+            jsonify(
+                {
+                    "success": False,
+                    "message": new_score
+                    + " does not fit within the expected bounds 0 < score < 1.0",
```
Comment (skrashevich) [inline frigate/http.py:395]: `<= 1.0`

**Comment by skrashevich** [comment]
I added `subLabelScore` support to the latest double-take beta build

File: frigate/http.py (hunk 1 above)
Comment (NickM-27) [reply frigate/http.py:395]: This is the case where we throw an error, a score of 100% or 0% is acceptable so it should not include those

Comment (skrashevich) [reply frigate/http.py:395]: > expected bounds 0 < score < 1.0
I understand this as the expected values are greater (not equal) than 0 and less than 1. I'm wrong?

Comment (NickM-27) [reply frigate/http.py:395]: by the way if you want to suggest code directly you can select code and use the suggest button

Screenshot 2023-07-10 at 6 57 05 PM

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**